packages/vm/tests/api/runBlock.spec.ts

```diff
@@ -229,6 +229,22 @@ tape('runBlock() -> API parameter usage/data errors', async (t) => {
       })
   })
 
+  t.test('should fail when no `validateHeader` method exists on blockchain class', async (t) => {
+    const vm = await VM.create({ common })
+    const blockRlp = toBuffer(testData.blocks[0].rlp)
+    const block = Object.create(Block.fromRLPSerializedBlock(blockRlp))
+    ;(vm.blockchain as any).validateHeader = undefined
```

Comment:
I have first tried to keep the blockchain interface at minimal since most methods are used somewhat internally and they are technically not necessary for a blockchain to support (e.g. if you dont have `genesisState` then only a small subset of features do not work). However, since you've gone forward and added these methods to the interface, then we should also remove those errors right, since now the interface specifies that these methods are necessary. (An implementer can add the method but always throw a `NotImplementedException`)

Reply:
Ah, I see.  Ugh, sorry, that makes a lot of sense.  Were you able to get the VM tests passing locally if we remove those other methods?

Reply:
I did not check if tests were passing since the tests did not compile at that point. But they should pass, I think we use our own blockchain object internally and not mock blockchains.

Reply:
At a minimum, we would need to add a check for that `validateHeader` like you did for `genesisState` if we want to remove that from the interface.  Is there anything else we could leave out of the interface without breaking the VM?

Reply:
I actually think we should keep `validateHeader` mandatory. The other methods, I'm not sure if we want to make those mandatory. Maybe let's not remove those from the interface but instead make them optional, i.e. add a `?` to them. I think all methods below `copy` in the interface except `validateHeader` should be made optional (and then we would indeed need those checks if we want to call that method!). This will also most likely break a lot of tests.